web/src/components/vis/PlsdaPage/PlsdaPage.vue

```diff
@@ -53,20 +56,24 @@ export default defineComponent({
 
     watchEffect(() => {
       const pcX = Number.parseInt(controls.pcXval, 10);
-      if (!Number.isNaN(pcX)) {
+      if (!Number.isNaN(pcX) && pcX <= controls.numComponents) {
```

Comment:
Did we ever arrive at some universal rule about when to use `Number.isNaN` vs. `window.isNaN`?

You're using the result of parseInt here, so it's not a problem, but in general these methods produce different outputs.  `Number.isNaN('asdf') === false` for example.

I think typescript protects us from that sort of nonsense, so it doesn't matter here.

Reply:
> You're using the result of parseInt here, so it's not a problem, but in general these methods produce different outputs. `Number.isNaN('asdf') === false` for example.

Oh goodness. TIL about this.

So, `window.isNaN(x)` seems roughly equivalent to `Number.isNaN(Number(x))`, is that right? That is, there are many values that convert to `NaN` when forced to a Number type, and `window.isNaN()` includes those semantics. If that's true, then I would say always use `Number.isNaN()`, and if you want the behavior of `window.isNaN()`, we should simply be explicit about it (by still using `Number.isNaN()` with an explicit conversion to `Number`).

But I literally just learned about this four minutes ago, so maybe there are corner cases or something.

Reply:
`Number.isNaN()` looks better to me, so that's what I prefer. I too was not aware of the distinction, but that seems like a convincing argument to me.

Reply:
`Number.isNaN()` is the most correct, I agree.